manifests/gem.pp

```diff
@@ -69,17 +69,22 @@
     fail('You must declare a ruby_version for rbenv::gem')
   }
 
+  $environment_for_install = concat(["RBENV_ROOT=${install_dir}"], $env)
+  $version_for_exec_name = regsubst($version, '[^0-9]+', '_', 'EG')
+
+  # Using shell if and substitution, build a test if rubygems is version 3 or
+  # greater.
+  $gem_ver3_test_command_prefix = 'IFS=. read -a gem_version_components <<< "$(gem --version)"; if [ "${gem_version_components[0]}" -ge 3 ]; then'
```

Comment:
Why don't we capture the version of rubygems in a variable and use a ternary switch inside `if ($skip_docs)` to use the correct way to no-doc?

Reply:
The problem here is that puppet does not inherently know the version of rubygems installed for any ruby version.  We could, theoretically, infer a default version for each rbenv version; but, that doesn't prevent people from doing a gem system update and breaking it (which happened in my environment when I tried that path).

The only way to get the rubygems version which maps to the current rbenv version you're working on would be a fact, since puppet wouldn't otherwise know at runtime.  The cleanest way would be a structured fact, such as a hash, which maps rbenv version designations to the resulting rubygems version.  We could then interrogate the fact to find the proper ruby version.

I saw one major drawback with this approach- any fact that gathered this information would have to loop on every version of ruby installed on the system and issue a `gem --version` command, spinning up a ruby interpreter just to garner the variable.  This would occur every puppet run, whether we used this variable or not.  This sounds trivial until you consider an environment with 100+ machines with many versions of ruby on each machine.  Then it becomes a fairly large overhead to issue on every puppet run for a variable we'd only use in a particular case.  (In my case, I have many servers running in a virtual environment with shared resources.  Thus, this is far from a zero cost.)

I landed on using shell builtins to achieve the same result (store the rubygems major version and if branch off of it).  Now, the rubygems version will only be determined if we specifically require it at runtime by the exec doing the work.